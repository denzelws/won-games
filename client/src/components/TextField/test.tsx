import { screen, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { renderWithTheme } from 'utils/tests/helpers'

import TextField from '.'

describe('<TextField />', () => {
  it('should render with label', () => {
    renderWithTheme(<TextField label="textfield" labelFor="field" />)

    expect(screen.getByLabelText(/textfield/i)).toBeInTheDocument()
  })

  it('should render without label', () => {
    renderWithTheme(<TextField />)

    expect(screen.queryByLabelText(/textfield/i)).not.toBeInTheDocument()
  })

  it('should render with placeholder', () => {
    renderWithTheme(<TextField placeholder="field check" />)

    expect(screen.getByPlaceholderText(/field check/i)).toBeInTheDocument()
  })

  it('changes its value when typing ', async () => {
    const onInput = jest.fn()
    renderWithTheme(
      <TextField
        onInput={onInput}
        label="textfield"
        labelFor="textfield"
        id="textfield"
      />
    )

    const input = screen.getByRole('textbox')
    const text = 'text example'

    userEvent.type(input, text)

    await waitFor(() => {
      expect(input).toHaveValue(text)
      expect(onInput).toHaveBeenCalledTimes(text.length)
    })

    expect(onInput).toHaveBeenCalledWith(text)
  })

  it('should be accessible with tab', async () => {
    renderWithTheme(
      <TextField label="textfield" labelFor="textfield" id="textfield" />
    )

    const input = screen.getByLabelText(/textfield/i)
    expect(document.body).toHaveFocus()

    await userEvent.tab()
    expect(input).toHaveFocus()
  })
})
