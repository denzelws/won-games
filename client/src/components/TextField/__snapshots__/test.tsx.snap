// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> should show an error message 1`] = `
.c0 .c2 {
  border-color: #FF6347;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #EAEAEA;
  border-radius: 0.2rem;
  padding: 0 1.6rem;
  border: 0.2rem solid;
  border-color: #EAEAEA;
}

.c3:focus-within {
  box-shadow: 0 0 0.5rem #F231A5;
}

.c4 {
  color: #030517;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.6rem;
  padding: 0.8rem;
  padding-left: 0.8rem;
  background: transparent;
  border: 0;
  outline: none;
  width: 100%;
}

.c5 {
  color: #FF6347;
  font-size: 1.6rem;
}

.c1 {
  font-size: 1.4rem;
  color: #030517;
  cursor: pointer;
}

<div
  class="c0"
>
  <label
    class="c1"
  >
    textfield
  </label>
  <div
    class="c2 c3"
  >
    <input
      class="c4"
      type="text"
      value=""
    />
  </div>
  <p
    class="c5"
  >
    Error message
  </p>
</div>
`;
